### PR TITLE
User error codes

### PR DESCRIPTION
**Pull request by wyb1**
**What this PR does / why we need it**:

Add the labels `has_errors` and `is_compliant` labels to the `garden_shoot_condition` metric. This will help with ignoring misconfigured shoots for alerting.

**Special notes for your reviewer**:
/cc @timuthy 

**Release note**:

```improvement operator
Upgrade gardener dependencies to v1.17.1
```

```improvement operator
Add `has_errors` label to `garden_shoot_condition` metric
```

```improvement operator
Add `is_compliant` label to `garden_shoot_condition` metric
```

### CODE REVIEW

**Review by istvanballok — APPROVED** [review]
Kudos!

File: pkg/metrics/shoot.go [hunk 1]
```diff
@@ -353,3 +361,12 @@ func (c gardenMetricsCollector) getSeeds() map[string]*gardenv1beta1.Seed {
 	}
 	return seeds
 }
+
+func shootIsCompliant(constraints *[]gardenv1beta1.Condition) string {
```
Comment (timuthy) [inline pkg/metrics/shoot.go:365]:
```suggestion
func shootIsCompliant(constraints []gardenv1beta1.Condition) string {
```
Nit: I think there is no benefit of using a pointer to an array.

File: pkg/metrics/metrics.go [hunk 2]
```diff
@@ -109,6 +109,8 @@ func getGardenMetricsDefinitions() map[string]*prometheus.Desc {
 				"seed_iaas",
 				"seed_region",
 				"uid",
+				"has_errors",
```
Comment (timuthy) [inline pkg/metrics/metrics.go:112]:
```suggestion
				"has_user_errors",
```
As suggested by @dguendisch

**Review by timuthy — CHANGES_REQUESTED** [review]
Thanks @wyb1. Only two suggestions from my side.

File: pkg/metrics/shoot.go (hunk 1 above)
Comment (wyb1) [reply pkg/metrics/shoot.go:365]: good point

**Pull request merged**